aptos-move/mvhashmap/src/versioned_aggregators.rs

```diff
@@ -128,18 +130,18 @@ impl VersionedValue {
                 if !match (&**o.get(), &entry) {
                     // These are the cases where the transaction behavior with respect to the
                     // aggregator may change (based on the information recorded in the Estimate).
-                    (Estimate(SnapshotBypass(id_l, delta_l)), Snapshot(id_r, delta_r)) => {
-                        *id_l == *id_r && *delta_l == *delta_r
-                    },
-                    (Estimate(DeltaBypass(delta_l)), Delta(delta_r) | Value(_, Some(delta_r))) => {
-                        *delta_l == *delta_r
+                    (Estimate(Bypass(apply_l)), Apply(apply_r) | Value(_, Some(apply_r))) => match (apply_l, apply_r) {
+                        (AggregatorDelta { .. }, AggregatorDelta { .. })
+                        | (SnapshotDelta { .. }, SnapshotDelta { .. })
+                        | (SnapshotDerived { .. }, SnapshotDerived { .. }) => {
+                            println!("Checking {:?} and {:?}, match: {}", apply_l, apply_r, *apply_l == *apply_r);
+                            *apply_l == *apply_r
+                        },
+                        _ => unreachable!(
```

Comment:
Checking, now that we merged everything into the same enum, is this guaranteed? I think so because IDs are always re-generated for re-executions and are atomic

Reply:
yes, IDs cannot ever be reused